clang/test/CIR/Lowering/ThroughMLIR/cos.cir

```diff
@@ -5,18 +5,26 @@ module {
   cir.func @foo() {
     %1 = cir.const(#cir.fp<1.0> : !cir.float) : !cir.float
     %2 = cir.const(#cir.fp<1.0> : !cir.double) : !cir.double
-    %3 = cir.cos %1 : !cir.float
-    %4 = cir.cos %2 : !cir.double
+    %3 = cir.const(#cir.fp<1.0> : !cir.long_double<!cir.f80>) : !cir.long_double<!cir.f80>
+    %4 = cir.const(#cir.fp<1.0> : !cir.long_double<!cir.double>) : !cir.long_double<!cir.double>
+    %5 = cir.cos %1 : !cir.float
+    %6 = cir.cos %2 : !cir.double
+    %7 = cir.cos %3 : !cir.long_double<!cir.f80>
+    %8 = cir.cos %4 : !cir.long_double<!cir.double>
     cir.return
   }
 }
 
-//CHECK: module {
-//CHECK:   func.func @foo() {
-//CHECK:     %cst = arith.constant 1.000000e+00 : f32
-//CHECK:     %cst_0 = arith.constant 1.000000e+00 : f64
-//CHECK:     %0 = math.cos %cst : f32
-//CHECK:     %1 = math.cos %cst_0 : f64
-//CHECK:     return
-//CHECK:   }
-//CHECK: }
+// CHECK: module {
+// CHECK:   func.func @foo() {
+// CHECK:     %cst = arith.constant 1.000000e+00 : f32
+// CHECK:     %cst_0 = arith.constant 1.000000e+00 : f64
+// CHECK:     %cst_1 = arith.constant 1.000000e+00 : f80
+// CHECK:     %cst_2 = arith.constant 1.000000e+00 : f64
+// CHECK:     %0 = math.cos %cst : f32
+// CHECK:     %1 = math.cos %cst_0 : f64
+// CHECK:     %2 = math.cos %cst_1 : f80
+// CHECK:     %3 = math.cos %cst_2 : f64
+// CHECK:     return
+// CHECK:   }
+// CHECK: }
```

Comment:
All `CHECK` directives except for the first should actually be `CHECK-NEXT`. And it's better to use [FileCheck substitution blocks](https://llvm.org/docs/CommandGuide/FileCheck.html#filecheck-string-substitution-blocks) so tests don't depend on value names:

```suggestion
//      CHECK: func.func @foo() {
// CHECK-NEXT:   %[[C0:.+]] = arith.constant 1.000000e+00 : f32
// CHECK-NEXT:   %[[C1:.+]] = arith.constant 1.000000e+00 : f64
// CHECK-NEXT:   %[[C2:.+]] = arith.constant 1.000000e+00 : f80
// CHECK-NEXT:   %[[C3:.+]] = arith.constant 1.000000e+00 : f64
// CHECK-NEXT:   %{{.+}} = math.cos %[[C0]] : f32
// CHECK-NEXT:   %{{.+}} = math.cos %[[C1]] : f64
// CHECK-NEXT:   %{{.+}} = math.cos %[[C2]] : f80
// CHECK-NEXT:   %{{.+}} = math.cos %[[C3]] : f64
// CHECK-NEXT:   return
// CHECK-NEXT: }
```

Reply:
Thanks for suggestion, I will modify